File: rest-client/src/main/java/org/acme/restclient/CountriesService.java
package org.acme.restclient;

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
import org.jboss.resteasy.annotations.jaxrs.PathParam;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import java.util.Set;
import java.util.concurrent.CompletionStage;

@Path("/v2")
@RegisterRestClient
public interface CountriesService {

    @GET
    @Path("/name/{name}")
    @Produces("application/json")
    Set<Country> getByName(@PathParam String name);

    @GET
    @Path("/name/{name}")
    @Produces("application/json")
    CompletionStage<Set<Country>> getByNameAsync(@PathParam String name);
}

File: rest-client-kotlin-quickstart/src/main/kotlin/org/acme/restclient/CountriesService.kt
package org.acme.restclient

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient
import org.jboss.resteasy.annotations.jaxrs.PathParam
import java.util.concurrent.CompletionStage
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces

@Path("/v2")
@RegisterRestClient
interface CountriesService {

    @GET
    @Path("/name/{name}")
    @Produces("application/json")
    fun getByName(@PathParam(value = "name") name: String): Set<Country>

    @GET
    @Path("/name/{name}")
    @Produces("application/json")
    fun getByNameAsync(@PathParam(value = "name") name: String): CompletionStage<Set<Country>>
}
